### PR TITLE
Upgrade sentry & set source maps

### PR DESCRIPTION
**Pull request by mattkrick**
# IMPORTANT `SENTRY_DSN` is now a public key. 
Before pushing to production, each server needs to:
- set `SENTRY_DSN = SENTRY_DSN_PUBLIC`
- remove `SENTRY_DSN_PUBLIC`
 
we've been on the legacy "raven" fro awhile now, figured fixing sentry sourcemaps was a good time to upgrade.

TEST
- [ ] source maps are provided to sentry when deploying (see sentry: https://sentry.io/parabol/action-production/releases/3.3.0-alpha.1/artifacts/)
- [ ] source maps are available in the console (see dev server)

notes: 
- we don't have to make them available on the client if we don't want to, it was just easier.
- a bunch of files were touched because i turned a couple files into typescript & it uncovered a few unused variables

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -162,6 +162,16 @@ jobs:
       - store_artifacts:
           path: ~/action/build
           destination: build
+      - run:
+          name: Upload sentry artifacts
+          command: |
+            if [ "${CIRCLE_BRANCH}" == "production" ]; then
+              curl -sL https://sentry.io/get-cli/ | bash
```
Comment (jordanh) [inline .circleci/config.yml]: Simple enough...they didn't have a Docker container mixin?

File: package.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
   "description": "An open-source app for building smarter, more agile teams.",
   "author": "Parabol Inc. <love@parabol.com> (http://github.com/ParabolInc)",
   "license": "AGPL-3.0",
-  "version": "3.2.0",
+  "version": "3.3.0-alpha.1",
```
Comment (jordanh) [inline package.json]: Ah... this is why the regex wasn't permissive enough!

File: webpack/webpack.prod.config.js [hunk 3]
```diff
@@ -86,9 +92,20 @@ module.exports = {
   optimization: {
     minimize: Boolean(process.env.WEBPACK_DEPLOY),
     minimizer: [
-      new UglifyJsPlugin({
-        exclude: /codemirror/,
-        parallel: true
+      new TerserPlugin({
```
Comment (jordanh) [inline webpack/webpack.prod.config.js]: Changing horses here, huh? Why?

**Comment by jordanh** [comment]
# Test Cycle 1

## Tests
- [ ] source maps are provided to sentry when deploying (see sentry: https://sentry.io/parabol/action-production/releases/3.3.0-alpha.1/artifacts/)
- [ ] source maps are available in the console (see dev server)

## Notes

File: .circleci/config.yml (hunk 1 above)
Comment (mattkrick) [reply .circleci/config.yml]: they do! want me to switch?

File: webpack/webpack.prod.config.js (hunk 3 above)
Comment (mattkrick) [reply webpack/webpack.prod.config.js]: this is the default in webpack now. the big change is uglify only works on es5, it's a little buggy, anda little slower. https://gist.github.com/mattkrick/6426ed85351b333b6408dbb7858b81c5

**Comment by mattkrick** [comment]
gonna leave this open until we publish so we can time the environment vars

**Comment by jordanh** [comment]
I'm all done testing here!

File: webpack/webpack.prod.config.js (hunk 3 above)
Comment (jordanh) [reply webpack/webpack.prod.config.js]: Cool!

File: .circleci/config.yml (hunk 1 above)
Comment (jordanh) [reply .circleci/config.yml]: Could you? It should speed deploys a little bit: CircleCI should cache the container

**Comment by mattkrick** [comment]
environments updated on staging & prod, they'll trigger on the next restart

**Pull request merged**